syft/pkg/cataloger/dotnet/licenses.go

```diff
@@ -0,0 +1,608 @@
+package dotnet
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/json"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/licenses"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/internal/fileresolver"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/scylladb/go-set/strset"
+)
+
+type nugetLicenseResolver struct {
+	opts                     CatalogerConfig
+	localNuGetCacheResolvers []file.Resolver
```

Comment:
When I initially read this, I thought there was some incorrect usage of file resolvers and wrote up some stuff about the differences between scan root and the host executing Syft. Then I realized you'd used these correctly with unindexed resolvers against the local filesystem, etc.. I don't think this is a blocker or necessarily needs to change at this point, but I did want to bring up that we've gone back and forth about this -- initially the golang licenses lookup code did the same thing; that's why the unindexed resolver was created, etc. but then we walked that decision back and [replaced the usage with standard `fs` package](https://github.com/anchore/syft/blob/main/syft/pkg/cataloger/golang/licenses.go#L44). Our latest discussions about it resulted in a decision to keep the `file.Resolver` _only_ for source content, and always use `fs` or `afero` for things we access on the host. So, if you think it would improve things to transition these to `fs.FS` or similar, and feel like doing the work, that would be great! But also, I think what you've done is fine here, too, but wanted to mention this since I'm sure you were not privy to any conversations about this.

Reply:
Thanks for the clarification about the decisions taken.
At the moment I am not quite sure, how I'd go about performing the switch towards the ```fs``` package...

Then again:
In a sense, the localNuGetCacheResolvers **are** source content:
These are the folders ('sources') from where the external NuGet dependencies are linked... 🤷‍♂️

@kzantow : What do you say?